src/login/AuthProviderBase.ts

```diff
@@ -24,8 +25,12 @@ export type AbstractCredentials2 = DeviceTokenCredentials2 | AzureIdentityCreden
 
 export abstract class AuthProviderBase<TLoginResult> {
 	private terminateServer: (() => Promise<void>) | undefined;
+	private handler: UriEventHandler = new UriEventHandler();
+
+	constructor() {
+		window.registerUriHandler(this.handler);
```

Comment:
Rather than move a bunch of code to this file, can we move it outside the auth provider classes? For example, you're now registering the uri handler twice (since we create an auth provider for both adal and msal), which at best is inefficient and at worst could cause bugs. It just doesn't seem like any of this code is unique to the auth providers

Reply:
Right now we're only [creating](https://github.com/microsoft/vscode-azure-account/blob/12ff3d1cdf34b511566565ccbf51fb1f086b8476/src/login/AzureLoginHelper.ts#L82-L84) one auth provider at a time but I agree this code might not belong here - I'll move it.